src/libaktualizr/primary/reportqueue.cc

```diff
@@ -112,6 +122,9 @@ void ReportQueue::flushQueue() {
       LOG_WARNING << "Failed to post update events: " << response.getStatusStr();
     }
     if (delete_events) {
+      if (report_array.size() < static_cast<size_t>(cur_event_number_limit_)) {
+        no_pending_events_ = true;
```

Comment:
I am not sure if this is needed or even correct. I think this condition does NOT mean that there is no any update events in the DB.

Reply:
It is fine to remove this. The only downside is doing an additional iteration before doing the `wait` without timeout.

Reply:
Yep, my understanding is that there are update events in the DB in this particular case, so ideally we should enforce a new iteration even without waiting with timeouts?

Reply:
I think the change is OK. My concern was related to this case https://github.com/foundriesio/aktualizr/blob/45f29eff34188583b13b65b684dccbcb94013191/src/libaktualizr/primary/reportqueue.cc#L110. I thought it is possible that this condition `if (report_array.size() < static_cast<size_t>(cur_event_number_limit_))` is executed after `cur_event_number_limit_` is divided by two within a single iteration.

Reply:
Yep, it should be fine. In that situation, `delete_events` should be `false`, so we won't hit the affected lines.